enumerables.rb

```diff
@@ -0,0 +1,130 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield to_a[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield to_a[i], i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    to_a.my_each { |item| result << item if yield(item) }
+    result
+  end
+
+  def my_all?(arg = nil)
+    to_a.my_each do |i|
+      if block_given?
+        return false if yield(i) == false
+      elsif arg
+        return true if arg == i
+      else
+        return true
+      end
+    end
+    false
+  end
+
+  def my_any?
+    return to_enum(:my_any?) unless block_given?
+
+    i = 0
+    ans = false
+    to_a.length.times do
+      ans = yield to_a[i]
+
+      break if ans == true
+
+      i += 1
+    end
+    ans
+  end
+
+  def my_none?
+    return to_enum(:my_none?) unless block_given?
+
+    i = 0
+    ans = false
+    to_a.length.times do
+      ans = yield to_a[i]
+
+      break if ans == true
+
+      i += 1
+    end
+    !ans
+  end
+
+  def my_count(number = nil)
+    count = 0
+    if block_given?
+      my_each { |i| count += 1 if yield i }
+    elsif number.nil?
+      count = to_a.length
+    else
+      to_a.my_each { |i| count += 1 if number == i }
+    end
+    count
+  end
+
+  def my_map(proc = nil)
```

Comment:
## my_map

in my_map this statement should be evaluated to true but it's false fix it.

```ruby

  array = [12, 3, 432, 52, 5, 2, 35, 2, 3, 23, 42, 52, 65, 68, 56]
  my_proc = proc { |num| num > 10 }

  array.my_map(my_proc) { |num| num < 10 } == array.my_map(&my_proc) => false

```